openedx/core/djangoapps/auth_exchange/views.py

```diff
@@ -27,24 +26,35 @@
 from openedx.core.djangoapps.oauth_dispatch.api import create_dot_access_token
 from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
 
-log = logging.getLogger(__name__)
+
+class CsrfExemptSessionAuthentication(SessionAuthentication):
+    """
+    Class enables CSRF exempt Session Authentication.
+    """
+    def enforce_csrf(self, request):
+        return  # Skip the CSRF check
 
 
 class AccessTokenExchangeBase(APIView):
     """
     View for token exchange from 3rd party OAuth access token to 1st party
     OAuth access token.
+
+    Note: This base class was originally created to support multiple libraries,
+        but we currently only support django-oauth-toolkit (DOT).
     """
-    @method_decorator(csrf_exempt)
+    # No CSRF protection is required because the provided 3rd party OAuth access
+    #  token is sufficient. However, we include session authentication to ensure
+    #  the session user and token user match, if there is a logged in user.
+    authentication_classes = [CsrfExemptSessionAuthentication]
+    allowed_methods = ['POST']
+
     @method_decorator(social_utils.psa("social:complete"))
     def dispatch(self, *args, **kwargs):  # pylint: disable=arguments-differ
         return super(AccessTokenExchangeBase, self).dispatch(*args, **kwargs)  # lint-amnesty, pylint: disable=super-with-arguments
 
     def get(self, request, _backend):
-        """
-        Pass through GET requests without the _backend
-        """
-        return super(AccessTokenExchangeBase, self).get(request)  # lint-amnesty, pylint: disable=no-member, super-with-arguments
+        return self._get_invalid_request_response('Only POST requests allowed.')
```

Comment:
Do we need both this and the `allowed_methods` attribute above?

Reply:
If you agree, I'm going to drop this method. This will switch to a 405 and would no longer be a JSON response. 

I had kept to the original JSON response to match the original implementation, which I thought matched our [api conventions](https://openedx.atlassian.net/wiki/spaces/AC/pages/18350757/edX+REST+API+Conventions), but it doesn't even include 405.

Reply:
I think drop it for now and do the default thing.